File: 27-march-2026/Frontend/src/src/services/todoService.js
import axios from "axios";

const BASE_URL = "http://localhost:1010/todos";

axios.interceptors.request.use((config) => {
  const token = localStorage.getItem("token");
  if (token) {
    config.headers.Authorization = `Bearer ${token}`;
  }
  return config;
});

export const getAllTodos = async () => {
  const response = await axios.get(BASE_URL);
  return response.data.data;
};

export const createTodo = async (todoData) => {
  const response = await axios.post(BASE_URL, todoData);
  return response.data.data;
};

export const deleteTodo = async (id) => {
  await axios.delete(`${BASE_URL}/${id}`);
};

export const getTodoById = async (id) => {
  const response = await axios.get(`${BASE_URL}/${id}`);
  return response.data.data;
};

export const updateTodo = async (id, todoData) => {
  const response = await axios.put(`${BASE_URL}/${id}`, todoData);
  return response.data.data;
};

File: 28-march-20226/todos/todos/backend/src/main/java/com/example/todos/service/TodoService.java
package com.example.todos.service;

import java.util.List;

import com.example.todos.dto.request.TodoRequest;
import com.example.todos.dto.response.TodoResponse;

public interface TodoService {
    TodoResponse createTodo(TodoRequest request, String username);
    List<TodoResponse> getAllTodos(String username);
    TodoResponse getTodoById(Long id, String username);
    TodoResponse updateTodo(Long id, TodoRequest request, String username);
    void deleteTodo(Long id, String username);
    TodoResponse toggleComplete(Long id, String username);
}